docs/testing/integration.md

```diff
@@ -0,0 +1,330 @@
+<!-- endSnippet -->
+
+SchemaName can not contain certain characters such as `-` and can not start with a number, so that's why it is not just a `Guid``.
+
+You can configure your host to use this scheme name like this:
+
+<!-- snippet: sample_integration_configure_scheme_name -->
+<a id='snippet-sample_integration_configure_scheme_name'></a>
+```cs
+Host = await AlbaHost.For<Program>(b =>
+{
+    b.ConfigureServices((context, services) =>
+    {
+        services.Configure<MartenSettings>(s =>
+        {
+            s.FromTests = true;
+            s.SchemaName = SchemaName;
+        });
+    });
+});
+```
+<sup><a href='https://github.com/JasperFx/marten/blob/master/src/Marten.AspNetCore.Testing/AppFixture.cs#L22-L34' title='Snippet source file'>snippet source</a> | <a href='#snippet-sample_integration_configure_scheme_name' title='Start of snippet'>anchor</a></sup>
+<!-- endSnippet -->
+
+`MartenSettings` is a custom config class, you can customize any way you'd like:
+
+<!-- snippet: sample_integration_settings -->
+<a id='snippet-sample_integration_settings'></a>
+```cs
+public class MartenSettings
+{
+    public const string SECTION = "Marten";
+    public string SchemaName { get; set; }
+    public bool FromTests { get; set; } = false;
+}
+```
+<sup><a href='https://github.com/JasperFx/marten/blob/master/src/IssueService/MartenSettings.cs#L3-L10' title='Snippet source file'>snippet source</a> | <a href='#snippet-sample_integration_settings' title='Start of snippet'>anchor</a></sup>
+<!-- endSnippet -->
+
+Now in your actual application you should configure the schema name:
+
+<!-- snippet: sample_integration_use_scheme_name -->
+<a id='snippet-sample_integration_use_scheme_name'></a>
+```cs
+services.AddMarten(sp =>
+{
+    var options = new StoreOptions();
+    options.Connection(ConnectionSource.ConnectionString);
+    var martenSettings = sp.GetRequiredService<IOptions<MartenSettings>>().Value;
+
+    if (!string.IsNullOrEmpty(martenSettings.SchemaName))
+    {
+        options.Events.DatabaseSchemaName = martenSettings.SchemaName;
+        options.DatabaseSchemaName = martenSettings.SchemaName;
+    }
+
+    return options;
+}).UseLightweightSessions();
+```
+<sup><a href='https://github.com/JasperFx/marten/blob/master/src/IssueService/Startup.cs#L32-L47' title='Snippet source file'>snippet source</a> | <a href='#snippet-sample_integration_use_scheme_name' title='Start of snippet'>anchor</a></sup>
+<!-- endSnippet -->
+
+::: warning
+Keep note that Marten can be configured to generate static code on startup that contains the scheme name, so it could be beneficial to turn that off for your integration tests:
+:::
+
+```cs
+var martenSettings = configuration.GetSection(MartenSettings.SECTION).Get<MartenSettings>() ?? new MartenSettings();
+if (martenSettings.FromTests)
```

Comment:
**QUESTION:** Why are you only applying database changes in test mode?

Also, my recommendation would be to always use `OptimizeArtifactWorkflow` as without that on CI you'll have a different configuration than in the regular deployment. Optimised Artefact Workflow will use dynamic generation for Development environment, so you should be fine to run tests locally without regenerating static types. That can be done on CI.

My suggestion would be to eliminate the `FromTests` setting and have the test settings the same as you'd have for production.

Reply:
The reasoning is to prevent code generation, since https://github.com/JasperFx/marten/discussions/2733 would be a problem if you run twice using two different schemes. That's why I do code generation in memory for integration tests.

If I understand correclty your alternative would be to use `ASPNETCORE_ENVIRONMENT` as `Development`. Which is default I think. I'm using `b.UseEnvironment("integration")` in my integration tests since I do not want appsettings.development.json in my tests.

Since I think most will use "development", I will make it a hint not to change that since code generation will not work with dynamic schemes.